src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -119,22 +119,7 @@ public CalledHaplotypes callMutations (
             final VariantContext mergedVC = AssemblyBasedCallerUtils.makeMergedVariantContext(eventsAtThisLoc);
             if( mergedVC == null ) { continue; }
 
-            // TODO: this varaible needs a descriptive name
-            final Map<VariantContext, Allele> mergeMap = new LinkedHashMap<>();
-
-            mergeMap.put(null, mergedVC.getReference()); // the reference event (null) --> the reference allele
-            for(int i = 0; i < eventsAtThisLoc.size(); i++) {
-                // TODO: as noted below, this operation seems dangerous. Understand how things can go wrong.
-                mergeMap.put(eventsAtThisLoc.get(i), mergedVC.getAlternateAllele(i)); // BUGBUG: This is assuming that the order of alleles is the same as the priority list given to simpleMerge function
-            }
-
-            /** TODO: the code in the for loop up to here needs refactor. The goal, as far as I can tell, is to create two things: alleleMapper and mergedVC
```

Comment:
The code used to require two `Map`s:
```java
final Map<Event, List<Haplotype>> eventMapper = createEventMapper(loc. . .
yada yada yada
final Map<VariantContext, Allele> mergeMap = new LinkedHashMap<>();
```
which were then joined to create a `Map<Allele, List<Haplotype>>`.  These `Map`s no longer exist.  I feel like the code is still not transparent enough, however.